ArtOfIllusion/src/artofillusion/math/Vec3.java

```diff
@@ -178,6 +178,20 @@ public final void normalize()
         z /= len;
       }
   }
+  
+  /** Crate a copy of this vector, scaled to the length of 1. If the length of this vector is 0, 
```

Comment:
Or perhaps you are starting to learn the Rust language? They put their code in crates... :D